src/features/currentTodo.ts

```diff
@@ -1,38 +1,33 @@
 import { Todo } from '../types/Todo';
 
-// we use string literal as a type to avoid mistype in future
 type RemoveTodoAction = { type: 'currentTodo/REMOVE' };
-
-// payload is a typical name for an action data
 type SetTodoAction = {
   type: 'currentTodo/SET';
   payload: Todo;
 };
+type State = Todo | null;
+type Action = SetTodoAction | RemoveTodoAction;
 
-// Action creator return type protect us from a mistype
 const removeTodo = (): RemoveTodoAction => ({ type: 'currentTodo/REMOVE' });
 
 const setTodo = (todo: Todo): SetTodoAction => ({
   type: 'currentTodo/SET',
   payload: todo,
 });
 
-// These actions will be used in the application
-export const actions = { setTodo, removeTodo };
-
-type State = Todo | null;
-type Action = SetTodoAction | RemoveTodoAction;
-
 const currentTodoReducer = (
   state: State = null,
   action: Action,
 ): State => {
   switch (action.type) {
-    // Implement all actions here
-
+    case 'currentTodo/SET':
+      return action.payload;
+    case 'currentTodo/REMOVE':
+      return null;
```

Comment:
move to constants `'currentTodo/SET'` and `'currentTodo/REMOVE'`

Reply:
@OlEzhA0 could you please explain what are the pros of such approach? 

In my variant I have typed action types and it protects from mistakes